openfl/component/collaborator/collaborator.py

```diff
@@ -184,8 +184,16 @@ def get_tasks(self):
     def do_task(self, task, round_number):
         """Do the specified task."""
         # map this task to an actual function name and kwargs
-        func_name = self.task_config[task]['function']
-        kwargs = self.task_config[task]['kwargs']
+        func_name = task.function_name
+
+        # There no enough information to understand that it's train or validate task
+        # Looks like we should provide task type.
+        kwargs = {}
```

Comment:
Removing the kwargs resolution from the plan will break the task runner API. Maybe add an if statement for this logic: 

```
    if hasattr(self.task_runner, 'TASK_REGISTRY'):
        func_name = task.function_name

        # There no enough information to understand that it's train or validate task
        # Looks like we should provide task type.
        kwargs = {}
        if func_name == 'validate':
            if task.is_local:
                kwargs['apply'] = 'local'
            else:
                kwargs['apply'] = 'global'
    else:
        func_name = self.task_config[task]['function']
        kwargs = self.task_config[task]['kwargs']
```

Reply:
Changed